packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -1,82 +1,313 @@
+        }
+
+        const attachmentFilename = this._allure.writeAttachment(content, contentType)
+
+        this.currentAllureSpec.addAttachment(
+            name,
+            {
+                contentType,
+            },
+            attachmentFilename
+        )
+    }
+
+    attachJSON(name: string, json: any) {
+        const isStr = typeof json === 'string'
+        const content = isStr ? json : JSON.stringify(json, null, 2)
+
+        this.attachFile(name, String(content), isStr ? ContentType.JSON : ContentType.TEXT)
+    }
+
+    attachScreenshot(name: string, content: Buffer) {
+        this.attachFile(name, content, ContentType.PNG)
+    }
+
+    _startSuite(suiteTitle: string) {
+        const newSuite: AllureGroup = this.currentSuite ? this.currentSuite.startGroup() : new AllureGroup(this._allure)
+
+        newSuite.name = suiteTitle
+
+        this._runningUnits.push(newSuite)
+    }
+
+    _endSuite() {
+        if (!this.currentSuite) {
+            throw new Error("There isn't any active suite!")
+        }
+
+        while (this.currentAllureSpec) {
+            const currentTest = this._runningUnits.pop() as AllureTest | AllureStep
+
+            if (currentTest instanceof AllureTest) {
+                currentTest.endTest()
+            } else {
+                currentTest.endStep()
+            }
+        }
+
+        const currentSuite = this._runningUnits.pop() as AllureGroup
+
+        currentSuite.endGroup()
+    }
+
+    _startTest(testTitle: string, cid?: string) {
+        const newTest = this.currentSuite ? this.currentSuite.startTest() : new AllureTest(this._allure)
+
+        newTest.name = testTitle
+
+        this._runningUnits.push(newTest)
+        this.setCaseParameters(cid)
+    }
+
+    _skipTest() {
+        if (!this.currentAllureSpec) {
+            return
+        }
+
+        const currentTest = this._runningUnits.pop() as AllureTest | AllureStep
+
+        currentTest.stage = Stage.PENDING
+        currentTest.status = AllureStatus.SKIPPED
+
+        if (currentTest instanceof AllureTest) {
+            currentTest.endTest()
+        } else {
+            currentTest.endStep()
+        }
+    }
+
+    _endTest(status: AllureStatus, error?: Error) {
+        if (!this.currentAllureSpec) {
+            return
+        }
+
+        const currentSpec = this._runningUnits.pop() as AllureTest | AllureStep
+
+        currentSpec.stage = Stage.FINISHED
+        currentSpec.status = status
+
+        if (error) {
+            currentSpec.detailsMessage = error.message
+            currentSpec.detailsTrace = error.stack
+        }
+
+        if (currentSpec instanceof AllureTest) {
+            currentSpec.endTest()
+        } else {
+            currentSpec.endStep()
+        }
+    }
+
+    _startStep(testTitle: string) {
+        if (!this.currentAllureSpec) {
+            throw new Error("There isn't any active test!")
+        }
+
+        const newStep = this.currentAllureSpec!.startStep(testTitle)
```

Comment:
the same as in https://github.com/webdriverio/webdriverio/pull/9704#discussion_r1112151894

Reply:
:+1: